integration/obscurogateway/obscurogateway_test.go

```diff
@@ -99,6 +105,119 @@ func TestObscuroGateway(t *testing.T) {
+	}
+
+	signedTx, err := w.SignTransaction(deployTx)
+	require.NoError(t, err)
+
+	err = user0.HTTPClient.SendTransaction(context.Background(), signedTx)
+	require.NoError(t, err)
+
+	contractReceipt, err := integrationCommon.AwaitReceiptEth(context.Background(), user0.HTTPClient, signedTx.Hash(), time.Minute)
+	require.NoError(t, err)
+
+	resultMessage, err := getStringValueFromSmartContractGetter(contractReceipt.ContractAddress, eventsContractABI, "message", user1.HTTPClient)
+	require.NoError(t, err)
+
+	// check if the value is the same as hardcoded in smart contract
+	hardcodedMessageValue := "foo"
+	assert.Equal(t, hardcodedMessageValue, resultMessage)
+
+	// subscribe with all three users for all events in deployed contract
+	var user0logs []types.Log
+	var user1logs []types.Log
+	var user2logs []types.Log
+	subscribeToEvents([]gethcommon.Address{contractReceipt.ContractAddress}, nil, user0.WSClient, &user0logs)
+	subscribeToEvents([]gethcommon.Address{contractReceipt.ContractAddress}, nil, user1.WSClient, &user1logs)
+	subscribeToEvents([]gethcommon.Address{contractReceipt.ContractAddress}, nil, user2.WSClient, &user2logs)
+
+	// user1 calls setMessage and setMessage2 on deployed smart contract with the account
+	// that was registered as the first in OG
+	user1MessageValue := "user1PrivateEvent"
+	// interact with smart contract and cause events to be emitted
+	_, err = integrationCommon.InteractWithSmartContract(user1.HTTPClient, user1.Wallets[0], eventsContractABI, "setMessage", "user1PrivateEvent", contractReceipt.ContractAddress)
+	require.NoError(t, err)
+	_, err = integrationCommon.InteractWithSmartContract(user1.HTTPClient, user1.Wallets[0], eventsContractABI, "setMessage2", "user1PublicEvent", contractReceipt.ContractAddress)
+	require.NoError(t, err)
+
+	// check if value was changed in the smart contract with the interactions above
+	resultMessage, err = getStringValueFromSmartContractGetter(contractReceipt.ContractAddress, eventsContractABI, "message", user1.HTTPClient)
+	require.NoError(t, err)
+	assert.Equal(t, user1MessageValue, resultMessage)
+
+	// user2 calls setMessage and setMessage2 on deployed smart contract with the account
+	// that was registered as the second in OG
+	user2MessageValue := "user2PrivateEvent"
+	// interact with smart contract and cause events to be emitted
+	_, err = integrationCommon.InteractWithSmartContract(user2.HTTPClient, user2.Wallets[1], eventsContractABI, "setMessage", "user2PrivateEvent", contractReceipt.ContractAddress)
+	require.NoError(t, err)
+	_, err = integrationCommon.InteractWithSmartContract(user2.HTTPClient, user2.Wallets[1], eventsContractABI, "setMessage2", "user2PublicEvent", contractReceipt.ContractAddress)
+	require.NoError(t, err)
+
+	// check if value was changed in the smart contract with the interactions above
+	resultMessage, err = getStringValueFromSmartContractGetter(contractReceipt.ContractAddress, eventsContractABI, "message", user1.HTTPClient)
```

Comment:
Can we use the same approach as the errors test ?
Something in the lines of :
```
pack, _ := errorsContractABI.Pack("force_require")
	_, err = ethStdClient.CallContract(context.Background(), ethereum.CallMsg{
		From: w.Address(),
		To:   &receipt.ContractAddress,
		Data: pack,
	}, nil)
	```

Reply:
changed